### PR TITLE
feat: add `vapor-unit-test` skills

### PR DESCRIPTION
**Pull request by noahchoii**
## Summary by CodeRabbit

* **Documentation**
  * Added new end-to-end unit testing guidelines for the monorepo using Vitest, React Testing Library, and related conventions (including coverage targets and test structure rules).
  * Added reference guides for React component tests, React hook tests, and general JS/TS testing.
  * Added example test suites for a Breadcrumb component (accessibility + keyboard behavior), a `useInterval` hook (fake timers + cleanup), and `stateful-props` utilities (merging/resolution behavior).

 ## Summary

Adds `vapor-unit-test` skill — convention guide for writing Vitest + React Testing Library
  unit tests across the vapor-ui monorepo. Triggered by phrases like "테스트 추가해줘", "write
  unit tests", "커버리지 보강", "test 좀 짜줘" — no explicit skill invocation required.

  ## Structure

  Progressive disclosure: skill body stays short, dispatches to a single reference per target.

```
  vapor-unit-test/
  ├── SKILL.md                          # routing + rules common to all targets
  └── references/
      ├── component-tests.md            # React components
      ├── hook-tests.md                 # custom hooks
      ├── general-tests.md              # pure JS/TS (functions, transforms, lint rules…)
      └── examples/
          ├── breadcrumb.test.tsx
          ├── use-hook.test.ts
          └── util.test.ts
```

  `SKILL.md` carries the routing table + rules that apply regardless of target (file
  placement, fixture layout, imports, naming, mocks, coverage). Each reference is loaded only
  when its target matches — keeps Claude's context lean.

  ## Decisions

  - **Three targets, three references** — component / hook / general JS-TS. Each has
  materially different assertion vocabulary (RTL queries vs `renderHook`/`act` vs
  input→output), and bundling them into one doc would force Claude to read irrelevant guidance
  on every invocation.
  - **Examples as separate files, not inline** — full runnable test files survive scrolling
  and let Claude pattern-match on real shapes.
  - **Co-location by default, `__tests__/` for cross-file flows** — keeps single-file targets
  discoverable; isolates multi-file integration scenarios where co-location would clutter the
  source folder.
  - **Fixtures live under `__testfixtures__/`** — single glob name (jscodeshift convention)
  used across all domains so Vitest/tsconfig can exclude them in one place; prevents
  partial-source fixtures from breaking build/typecheck.
  - **Coverage gate: 80% across four indicators** — single explicit threshold, with a
  delegation rule (one-line forward to external lib = skip) so the guide doesn't push toward
  testing third-party code.
  - **TDD optional** — primary use case is writing tests against existing code; forcing
  red→green→refactor would conflict with how the skill is actually invoked.
  - **Visual props excluded** — `size`/`variant`/`color`/`shape` belong to visual regression;
  declaring this up front prevents redundant unit tests.

## Checklist

Before submitting the PR, please make sure you have checked all of the following items.

- [x] The PR title follows the Conventional Commits convention. (e.g., feat, fix, docs, style, refactor, test, chore)
- [ ] I have added tests for my changes.
- [ ] I have updated the Storybook or relevant documentation.
- [ ] I have added a changeset for this change. (e.g., for any changes that affect users, such as component prop changes or new features).
- [ ] I have performed a self-code review.
- [x] I have followed the project's [coding conventions](https://github.com/goorm-dev/vapor-ui/blob/main/.gemini/styleguide.md) and component patterns.

### CODE REVIEW

File: .claude/skills/vapor-unit-test/references/component-tests.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+컴포넌트 이름을 문자열로 사용합니다.
+
+```tsx
+describe('Button', () => {
+    // ...
+});
+```
+
+`cleanup`은 RTL이 자동으로 호출하므로 **기본적으로 추가하지 않습니다.** 한 `it()` 내부에서 여러 번 `render`를 호출하는 등 특수한 경우에만 명시적으로 `afterEach(cleanup)` 또는 `cleanup()`을 직접 호출하세요.
```
Comment (MaxLee-dev) [inline .claude/skills/vapor-unit-test/references/component-tests.md]: In this case, rather than manually calling cleanup within a single test, I think it would be more appropriate to separate the test cases.

Therefore, instead of guiding the use of cleanup as an example for special cases, how about updating the convention to recommend using rerender or separating the test cases?

Additionally, for edge cases where manual DOM cleanup is actually required, unmount is generally used much more often than cleanup. Since we are writing component-level tests, there will be scenarios where we need to test specific behaviors upon unmounting (e.g., event listener removal for a Sheet or Dialog). For these situations, specifically calling unmount() on the target component is much more desirable than using cleanup() to wipe out the entire global DOM!

File: .claude/skills/vapor-unit-test/references/component-tests.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+## describe 그룹
+
+테스트가 **2개 이상의 카테고리**로 나뉠 때만 nested describe로 묶습니다. 카테고리가 1개거나 테스트가 몇 개 안 되면 nested 없이 최상위 `describe`에 평탄하게 둡니다 — 불필요한 중첩은 가독성을 해칩니다.
```
Comment (MaxLee-dev) [inline .claude/skills/vapor-unit-test/references/component-tests.md]: The phrasing in "카테고리가 1개거나 테스트가 몇 개 안 되면 nested 없이" feels a bit ambiguous.

How about updating it to "카테고리가 1개거나, 테스트 케이스의 케이스가 적어 카테고리를 분류할 수 없는 경우 nested 없이" for better clarity?

**Review by MaxLee-dev — CHANGES_REQUESTED** [review]
I understand that the content you've provided serves as a convention for test cases. For this type of content, Anthropic officially recommends adding it under `rules`.

Since `skills` are only triggered based on the conditions defined in their descriptions, how about moving it to [[rules](https://code.claude.com/docs/ko/memory#organize-rules-with-claude%2Frules%2F)], which are triggered based on file paths? Please let me know what you think!

File: .claude/skills/vapor-unit-test/references/component-tests.md (hunk 1 above)
Comment (noahchoii) [reply .claude/skills/vapor-unit-test/references/component-tests.md]: cleanup은 내부적으로 `unmount()`를 포함하고 있기도 하고, 많이 쓰는 패턴으로 알고 있긴 합니다..! 그리고 `afterEach(cleanup)`이 의미가 있었던 이유는 매번 동일한 테스트베드를 호출하는 번거로움을 줄이기 위해 아래와 같이 작성하고 있기 때문이었습니다!

```tsx
describe('xxx', () => {
    let rendered: RenderResult;

    beforeEach(() => {
        rendered = render(<TestBed />);
    })

    it('can test without render testbed', () => {
        const element = rendered.getByRole('textbox');
        // ...
    })
})
```

이 때 사전에 생성된 테스트베드 대신 새로운 케이스를 띄워야 하는 경우 DOM을 지우고 새롭게 render하기 위해 cleanup을 호출해야 했어요.

다만 제가 놓친 부분이 있어서 정리해서 제안드리자면 다음과 같습니다!

1. vitest 설정 파일에 `globals: true`를 지정해주면 RTL 같은 라이브러리에 의해 DOM이 auto-cleanup 된다고 하네요! 그래서 `afterEach(cleanup)`은 더이상 필요없을 것 같아요.
2. 그리고 제안해주신 특수 케이스에 대한 사용을 제거하기 위해서는 `rendered = beforeEach()`와 같은 사용법을 지양해야 하는데, 조금 번거롭긴 하겠지만 그래도 각 테스트케이스에서 사용할 테스트베드를 직접 설정해준다는 의미로 해당 가이드는 제거해두도록 하겠습니다!
